src/commands/search.js

```diff
@@ -148,11 +148,12 @@ class SearchCommand extends BoxCommand {
 
 		let results = await this.client.search.query(args.query || null, options);
 
-		// Hard limit the search results to avoid slamming the API
+		// Limit the search results according to the --limit flag value (if specified) or RESULTS_LIMIT value
+		const itemsLimit = flags.limit || RESULTS_LIMIT;
```

Comment:
Don't you also need to modify line 30 where `options.limit` is set to use `flags.limit` instead of the hardcoded `RESULTS_LIMIT`?

Reply:
Actually no. We don't want to use limit parameter the same as API understands that. For API limit means number of items per page, but CLI makes pagination for us. So in our case limit means total number of items returned by CLI. You can also read this thread: https://box.slack.com/archives/C6KSD7C91/p1651243355904129